core/src/main/java/com/linecorp/armeria/common/logging/DefaultRequestLog.java

```diff
@@ -109,6 +113,121 @@ public DefaultRequestLog(RequestContext ctx) {
         this.ctx = requireNonNull(ctx, "ctx");
     }
 
+    @Override
+    public void addChild(RequestLog child) {
+        checkState(!hasLastChild, "last child is already added");
+        requireNonNull(child, "child");
+        if (children == null) {
+            // first child's all request-side logging events are propagated immediately to the parent
+            children = new ArrayList<>();
+            propagateRequestSideLog(child);
+        }
+        children.add(child);
+    }
+
+    private void propagateRequestSideLog(RequestLog child) {
+        child.addListener(log -> {
+            final long requestStartTimeNanos;
+            final long requestStartTimeMillis;
+            if (log instanceof DefaultRequestLog) {
```

Comment:
nit: How about adding `requestStartTimeNanos()` and `requestEndTimeNanos()` to `RequestLog` interface? @trustin WDYT?

Reply:
That was my first thought. But I was not sure if it's ok to expose the value to the outside since it is relative value.
I'm not sticking to that though :-).